src/Controller/HomeController.php

```diff
@@ -19,4 +24,96 @@ public function index(TrickRepository $trickRepository)
             'tricks' => $tricks
         ]);
     }
+
+    /**
+     * @Route("/trick/create", name="trick_add")
+     */
+    public function trickAdd(Request $request, EntityManagerInterface $manager)
+    {
+        $trick= new Trick();
+        $media = new Media();
+        $trick->addMedia($media);
+        $form = $this->createForm(TrickType::class, $trick);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+
+            $this->setParsedMedias($trick);
+            $trick->setAuthor($this->getUser());
+            $trick->setSlug($trick->getSlug());
```

Comment:
still useful ?

Reply:
yes it is actually

Reply:
Indeed $trick->setSlug($trick->getSlug()); wasn't necessary anymore cause it's EventListener which does it. I deleted it. 
Others lines are ok.